cookiecutter/zipfile.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import
+
+import os
+import requests
+import sys
+from zipfile import ZipFile
+
+from .prompt import read_user_yes_no
+from .utils import make_sure_path_exists, rmtree
+
+
+def prompt_and_delete(path, no_input=False):
+    """Ask the user whether it's okay to delete the previously-downloaded
+    file/directory.
+
+    If yes, deletes it. Otherwise, Cookiecutter exits.
+
+    :param path: Previously downloaded zipfile.
+    :param no_input: Suppress prompt to delete repo and just delete it.
+    """
+    # Suppress prompt if called via API
+    if no_input:
+        ok_to_delete = True
+    else:
+        question = (
+            "You've downloaded {} before. "
+            "Is it okay to delete and re-download it?"
+        ).format(path)
+
+        ok_to_delete = read_user_yes_no(question, 'yes')
+
+    if ok_to_delete:
+        if os.path.isdir(path):
+            rmtree(path)
+        else:
+            os.remove(path)
+    else:
+        sys.exit()
+
+    return ok_to_delete
+
+
+def unzip(zip_url, is_url, clone_to_dir='.', no_input=False):
+    # Ensure that clone_to_dir exists
+    clone_to_dir = os.path.expanduser(clone_to_dir)
+    make_sure_path_exists(clone_to_dir)
+
+    if is_url:
+        # Build the name of the cached zipfile,
+        # and prompt to delete if it already exists.
+        identifier = zip_url.rsplit('/', 1)[1]
+        zip_path = os.path.join(clone_to_dir, identifier)
```

Comment:
I did think about this; I ended up keeping both on the "what's a few more bytes?" principle. Effectively, we have three options:
1. Keep both (what is currently implemented)
2. Keep the zip file, and unpack to a temp directory
3. Keep the unpacked victory, and delete the zip file

Moving to (2) is a simple solution; but (3) gives us the opportunity (at some point in the future) use a password encoded zip file, adding to the potential security for commercial templates. 

So - any preferences?